src/ucs/config/parser.c

```diff
@@ -1437,13 +1464,8 @@ ucs_config_parser_set_value_internal(void *opts, ucs_config_field_t *fields,
                     return status;
                 }
             }
-        } else if (/* check exact match */
-                   (((table_prefix == NULL) ||
-                     !strncmp(name, table_prefix, prefix_len)) &&
-                    !strcmp(name + prefix_len, field->name)) ||
-                   /* check for wildcard */
-                   !fnmatch(name, field->name, 0))
-        {
+        } else if (ucs_config_prefix_name_match(table_prefix, prefix_len,
+                                                field->name, name)) {
```

Comment:
To simplify the code, WDYT?

```c
else if (((prefix_len == 0) && !fnmatch(name, field->name)) ||
         ucs_config_prefix_name_match(table_prefix, prefix_len,
                                      field->name, name)) {
}
```

and rename ucs_config_prefix_name_match_no_check to ucs_config_prefix_name_match

Reply:
I think it does not look like simplifucation, in this case:
1) name matching ligic is brocken up in two places
2) it complecates entire `ucs_config_parser_set_value_internal` function and `if () else if ()` statements